development/wasm-filter-plugins.md

```diff
@@ -0,0 +1,162 @@
+# WASM Filter Plugins
+
+Fluent Bit currently supports integration of wasm plugins built as wasm/wasi objects for input and filter plugins only. The interface for the WASM filter plugins is currently under development but is functional.
+
+## Prerequisites
+
+### For Build fluent-bit
+
+For only execution of WASM input and filter plugins, there is no additional requirement.
+
+#### Building flb-wamrc (optional)
+
+For `flb-wamrc` support, users have to install llvm infrastructure, e.g:
+
+```text
+# apt install -y llvm
+```
+
+### For Build WASM programs
+
+Currently, fluent-bit supports the following WASM toolchains:
+
+* Rust on `wasm32-unknown-unknown`.
+  * rustc 1.62.1 (e092d0b6b 2022-07-16) or later
+* [TinyGo](https://github.com/tinygo-org/tinygo) on wasm32-wasi
+  * v0.24.0 or later
+* [WASI SDK](https://github.com/WebAssembly/wasi-sdk) 13 or later.
+
+## Getting Started
+
+Compile Fluent Bit with WASM support, e.g:
```

Comment:
Can we just reference the other docs on building?

Reply:
AFAIK, I'd never seen the complete documentation to build fluent-bit in other fluent-bit doc.

Reply:
we should add it under this list https://docs.fluentbit.io/manual/installation/sources/build-and-install#build-options

Reply:
Added in https://github.com/fluent/fluent-bit-docs/pull/942/commits/69ef376bf0c072db422c237e4ba741ccab971de8